supervisor/docker/homeassistant.py

```diff
@@ -53,10 +55,15 @@ def name(self) -> str:
     @property
     def timeout(self) -> int:
         """Return timeout for Docker actions."""
-        # Synchronized with the homeassistant core container's S6_SERVICES_GRACETIME
-        # to avoid killing Home Assistant Core, see
+        # Use S6_SERVICES_GRACETIME to avoid killing Home Assistant Core, see
         # https://github.com/home-assistant/core/tree/dev/Dockerfile
-        return 240 + 20
+        if self.meta_config and "Env" in self.meta_config:
+            for env in self.meta_config["Env"]:
+                if match := ENV_S6_GRACETIME.match(env):
+                    return 20 + int(int(match.group(1)) / 1000)
+
+        # Fallback - as of 2024.3, S6 SERVICES_GRACETIME was set to 24000
+        return 260
```

Comment:
Just realized that this fallback also comes into play for the landing page. It is built from scratch, so won't have any of the S6 environment variables set. 260 is probably a bit high, but also shouldn't matter really.

However, what I wonder is will the timeout be updated once the Core got downloaded? :thinking: